### PR TITLE
[WIP] Refactor cleanup moving code from header files to impl files

### PR DESCRIPTION
**Pull request by rommelDB**
Fixes partially #1131

### CODE REVIEW

**Comment by williamBlazing** [comment]
rerun-tests

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# BlazingSQL 0.18.0 (Date TBS)
+
+
+## New Features
+
+
+## Improvements
+- #1166 Moving code from header files to implementation files
```
Comment (williamBlazing) [inline CHANGELOG.md:8]: please update the PR number

File: engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include "CacheMachine.h"
 #include "io/Schema.h"
 #include "utilities/CommonOperations.h"
+#include "taskflow/executor.h"
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h:9]: This file in general needs to be refactored as well

**Review by williamBlazing — CHANGES_REQUESTED** [review]

**Pull request merged**